django/test/runner.py

```diff
@@ -5,30 +5,226 @@
+                # noisy.  The -Wd and -Wa flags can be used to bypass this
+                # only when self.warnings is None.
+                if self.warnings in ['default', 'always']:
+                    warnings.filterwarnings(
+                        'module', category=DeprecationWarning, message=r'Please use assert\w+ instead.')
+            startTime = time.time()
+            startTestRun = getattr(result, 'startTestRun', None)
+            if startTestRun is not None:
+                startTestRun()
+            try:
+                test(result)
+            finally:
+                stopTestRun = getattr(result, 'stopTestRun', None)
+                if stopTestRun is not None:
+                    stopTestRun()
+            stopTime = time.time()
+        timeTaken = stopTime - startTime
+        result.printErrors()
+        if hasattr(result, 'separator2'):
+            self.stream.writeln(result.separator2)
+        run = result.testsRun
+        self.stream.writeln("Ran {} test{} in {}s".format(
+            style.TEST_INFO(run), run != 1 and "s" or "", style.TEST_INFO("{:.3f}".format(timeTaken))))
+        self.stream.writeln()
+
+        expectedFails = unexpectedSuccesses = skipped = 0
+        try:
+            results = map(len, (result.expectedFailures,
+                                result.unexpectedSuccesses,
+                                result.skipped))
+        except AttributeError:
+            pass
+        else:
+            expectedFails, unexpectedSuccesses, skipped = results
+
+        infos = []
+        if not result.wasSuccessful():
+            self.stream.write(style.TEST_FAIL("FAILED"))
+            self.stream.write(result.terminator)
+            failed, errored = len(result.failures), len(result.errors)
+            if failed:
+                infos.append("failures={}{}".format(style.TEST_FAIL(failed), result.terminator))
+            if errored:
+                infos.append("errors={}{}".format(style.TEST_ERROR(errored), result.terminator))
+        else:
+            self.stream.write(style.TEST_SUCCESS("OK"))
+            self.stream.write(result.terminator)
+        if skipped:
+            infos.append("skipped={}".format(style.TEST_INFO(skipped)))
+        if expectedFails:
+            infos.append("expected failures={}{}".format(
+                style.TEST_EXPECTED_FAILURE(expectedFails), result.terminator))
+        if unexpectedSuccesses:
+            infos.append("unexpected successes={}{}".format(
+                style.TEST_UNEXPECTED_SUCCESS(unexpectedSuccesses), result.terminator))
+        if infos:
+            self.stream.writeln(" (%s)" % (", ".join(infos),))
+        else:
+            self.stream.write("\n")
+        return result
+
+    def _makeResult(self):
+        return self.resultclass(self.stream, self.descriptions, self.verbosity, no_color=self.no_color)
+
+
+def get_sql_highlighter(bg='light'):
```

Comment:
Can we move this function up by `get_traceback_highlighter`. (That was we keep the two related helper functions next to each other.)

Reply:
I think current function location is better because it's closer to where it used.

Reply:
😃 Yeah, I can see that. 

But I was scanning the code and looking for the definition. I want to find all the helpers in one place. 

(As ever, there are different ways of wanting to read it. That's OK.) 

If you feel strongly you can leave it for now. Tim will have a look before merging it anyway so he can input.

Reply:
No problem, I will move it.